central/resourcecollection/service/service_impl.go

```diff
@@ -83,3 +89,64 @@ func (s *serviceImpl) getCollection(ctx context.Context, id string) (*v1.GetColl
 		Deployments: nil,
 	}, nil
 }
+
+// CreateCollection creates a new collection from the given request
+func (s *serviceImpl) CreateCollection(ctx context.Context, request *v1.CreateCollectionRequest) (*v1.CreateCollectionResponse, error) {
+	if !features.ObjectCollections.Enabled() {
+		return nil, nil
+	}
+
+	if request.GetName() == "" {
+		return nil, errors.Wrap(errox.InvalidArgs, "Collection name should not be empty")
+	}
+	// check if collection with same name doesn't already exist
+	nameQuery := search.NewQueryBuilder().AddExactMatches(search.CollectionName, request.GetName()).ProtoQuery()
+	c, err := s.datastore.Count(ctx, nameQuery)
+	if err != nil {
+		return nil, err
+	}
+	if c != 0 {
+		return nil, errors.Wrap(errox.AlreadyExists, "A collection with that name already exists")
+	}
+
+	creator := extractUserIdentity(ctx)
+	if creator == nil {
+		return nil, errors.New("User identity not provided")
+	}
+
+	collection := &storage.ResourceCollection{
+		Id:                uuid.NewV4().String(),
+		Name:              request.GetName(),
+		Description:       request.GetDescription(),
+		CreatedAt:         protoconv.ConvertTimeToTimestamp(time.Now()),
+		CreatedBy:         creator,
+		ResourceSelectors: request.GetResourceSelectors(),
+	}
+
+	if len(request.GetEmbeddedCollectionIds()) > 0 {
+		embeddedCollections := make([]*storage.ResourceCollection_EmbeddedResourceCollection, 0, len(request.GetEmbeddedCollectionIds()))
+		for _, id := range request.GetEmbeddedCollectionIds() {
+			embeddedCollections = append(embeddedCollections, &storage.ResourceCollection_EmbeddedResourceCollection{Id: id})
+		}
+		collection.EmbeddedCollections = embeddedCollections
+	}
+
+	err = s.datastore.AddCollection(ctx, collection)
+	if err != nil {
+		return nil, err
+	}
+
+	return &v1.CreateCollectionResponse{Collection: collection}, nil
+}
+
+func extractUserIdentity(ctx context.Context) *storage.SlimUser {
+	ctxIdentity := authn.IdentityFromContextOrNil(ctx)
+	if ctxIdentity == nil {
+		return nil
+	}
+
+	return &storage.SlimUser{
+		Id:   ctxIdentity.UID(),
+		Name: ctxIdentity.FullName(),
```

Comment:
@keyallis @md2119 
Identity  has FullName and FriendlyName among other things. Is it correct to set the name to FullName here?